Project: Investigate a Dataset (TMDb movie data)
¶
Table of Contents
¶


Introduction


Data Wrangling


Explore and Visuals


Studying Genre


Studying Budget


Studying Revenue


Studying Vote Average


Relations


Conclusions


Limitations

Introduction
¶
This project is a part of a udacity nanodegree program to be a Data Analyst.

    This data has been collected from this source (
https://www.kaggle.com/tmdb/tmdb-movie-metadata
).




Going throu this project asking questions as a way to analyse the data in an appropriate way.       
The goals in this project are: 

Studying Genre 
Srudying Budget 
Studying Revenue 
Studying Rate (vote average)

Knowing the relation between rate and budget 
Knowing the relation between rate and revenue




Questions:




Q- What is the most frequent genre in the data

Q- What are the top 10 movies according to Budget
Q- What is the highest Budget 

Q- What are the top 10 movies according to Revenue
Q- What is the highest Revenue

Q- What are top 10 movies according to Rate
Q- What is the highest title rate
Q- What is the rate categories
Q- Who is the highest rated director

Q- How Budget affect Rate
Q- How Revenue affect Rate

Install Libraries
¶

In [ ]:
import pandas as pd
import numpy as np 
import matplotlib.pyplot as plt
import seaborn as sns 
%matplotlib inline

Defined Functions
¶

In [ ]:
def scatter_plot(df, x, y, title= 'title', xlabel= 'xlabel', ylabel= 'ylabel', color= 'color'):
    df.plot.scatter(x= x,
                    y= y, 
                    figsize = (8, 8), 
                    color= color
                   )
    plt.title(title)
    plt.xlabel(xlabel)
    plt.ylabel(ylabel)
    plt.show()

def scatter_plot_index(df, x, y, title= 'title', xlabel= 'xlabel', ylabel= 'ylabel'):
    plt.scatter(x= df[x], 
                y= df[y].index
               )
    plt.title(title), 
    plt.xlabel(xlabel), 
    plt.ylabel(ylabel)
    plt.show()
    
def top10_diagram(column, x, y, title= 'title', xlabel= 'xlabel', ylabel= 'ylabel'):
    df.nlargest(10, column).plot(
                                    x = x, 
                                    y= y, 
                                    kind= 'barh', 
                                    title = title, 
                                    xlabel = xlabel, 
                                    ylabel = ylabel, 
                                    figsize = (10, 6),
                                    color = '#2a6099',
                                    );

Data Wrangling
¶

Upload Files
¶

In [ ]:
df = pd.read_csv('tmdb-movies.csv')

Explore Data

In [ ]:
df

Out[4]:
















id


imdb_id


popularity


budget


revenue


original_title


cast


homepage


director


tagline


...


overview


runtime


genres


production_companies


release_date


vote_count


vote_average


release_year


budget_adj


revenue_adj










0


135397


tt0369610


32.985763


150000000


1513528810


Jurassic World


Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...


http://www.jurassicworld.com/


Colin Trevorrow


The park is open.


...


Twenty-two years after the events of Jurassic ...


124


Action|Adventure|Science Fiction|Thriller


Universal Studios|Amblin Entertainment|Legenda...


6/9/15


5562


6.5


2015


1.379999e+08


1.392446e+09






1


76341


tt1392190


28.419936


150000000


378436354


Mad Max: Fury Road


Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...


http://www.madmaxmovie.com/


George Miller


What a Lovely Day.


...


An apocalyptic story set in the furthest reach...


120


Action|Adventure|Science Fiction|

In [ ]:
df.head()

Out[5]:
















id


imdb_id


popularity


budget


revenue


original_title


cast


homepage


director


tagline


...


overview


runtime


genres


production_companies


release_date


vote_count


vote_average


release_year


budget_adj


revenue_adj










0


135397


tt0369610


32.985763


150000000


1513528810


Jurassic World


Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...


http://www.jurassicworld.com/


Colin Trevorrow


The park is open.


...


Twenty-two years after the events of Jurassic ...


124


Action|Adventure|Science Fiction|Thriller


Universal Studios|Amblin Entertainment|Legenda...


6/9/15


5562


6.5


2015


1.379999e+08


1.392446e+09






1


76341


tt1392190


28.419936


150000000


378436354


Mad Max: Fury Road


Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...


http://www.madmaxmovie.com/


George Miller


What a Lovely Day.


...


An apocalyptic story set in the furthest reach...


120


Action|Adventure|Science Fiction|

In [ ]:
df.columns

Out[6]:




Index(['id', 'imdb_id', 'popularity', 'budget', 'revenue', 'original_title',
       'cast', 'homepage', 'director', 'tagline', 'keywords', 'overview',
       'runtime', 'genres', 'production_companies', 'release_date',
       'vote_count', 'vote_average', 'release_year', 'budget_adj',
       'revenue_adj'],
      dtype='object')

In [ ]:
df.shape

Out[7]:




(10866, 21)

In [ ]:
df.describe()

Out[8]:
















id


popularity


budget


revenue


runtime


vote_count


vote_average


release_year


budget_adj


revenue_adj










count


10866.000000


10866.000000


1.086600e+04


1.086600e+04


10866.000000


10866.000000


10866.000000


10866.000000


1.086600e+04


1.086600e+04






mean


66064.177434


0.646441


1.462570e+07


3.982332e+07


102.070863


217.389748


5.974922


2001.322658


1.755104e+07


5.136436e+07






std


92130.136561


1.000185


3.091321e+07


1.170035e+08


31.381405


575.619058


0.935142


12.812941


3.430616e+07


1.446325e+08






min


5.000000


0.000065


0.000000e+00


0.000000e+00


0.000000


10.000000


1.500000


1960.000000


0.000000e+00


0.000000e+00






25%


10596.250000


0.207583


0.000000e+00


0.000000e+00


90.000000


17.000000


5.400000


1995.000000


0.000000e+00


0.000000e+00






50%


20669.000000


0.383856


0.000000e+00


0.000000e+00


99.000000


38.000000


6.000000


2006.000000


0

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10866 entries, 0 to 10865
Data columns (total 21 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   id                    10866 non-null  int64  
 1   imdb_id               10856 non-null  object 
 2   popularity            10866 non-null  float64
 3   budget                10866 non-null  int64  
 4   revenue               10866 non-null  int64  
 5   original_title        10866 non-null  object 
 6   cast                  10790 non-null  object 
 7   homepage              2936 non-null   object 
 8   director              10822 non-null  object 
 9   tagline               8042 non-null   object 
 10  keywords              9373 non-null   object 
 11  overview              10862 non-null  object 
 12  runtime               10866 non-null  int64  
 13  genres                10843 non-null  object 
 14  production_companies  9836 non-null   object 
 15  release_date       

Cleaning Data
¶

Drop Columns
¶

In [ ]:
# drop unused columns, change date data type
df.drop(['homepage', 'tagline', 'keywords', 'production_companies', 'cast', 'overview'], axis=1, inplace=True)
df['release_date']= pd.to_datetime(df['release_date'])
df.head()

Out[10]:
















id


imdb_id


popularity


budget


revenue


original_title


director


runtime


genres


release_date


vote_count


vote_average


release_year


budget_adj


revenue_adj










0


135397


tt0369610


32.985763


150000000


1513528810


Jurassic World


Colin Trevorrow


124


Action|Adventure|Science Fiction|Thriller


2015-06-09


5562


6.5


2015


1.379999e+08


1.392446e+09






1


76341


tt1392190


28.419936


150000000


378436354


Mad Max: Fury Road


George Miller


120


Action|Adventure|Science Fiction|Thriller


2015-05-13


6185


7.1


2015


1.379999e+08


3.481613e+08






2


262500


tt2908446


13.112507


110000000


295238201


Insurgent


Robert Schwentke


119


Adventure|Science Fiction|Thriller


2015-03-18


2480


6.3


2015


1.012000e+08


2.716190e+08






3


140607


tt2488496


11.173104


200000000


2068178225


Star Wars: The Force Awakens


J.J. Abrams


136


Action|Adventure|Science Fiction|Fantasy


20

In [ ]:
#Drop null values and check
df.dropna(inplace=True)
df.isnull().sum().any()

Out[11]:




False

In [ ]:
# Drop duplicated values then check
df.drop_duplicates(inplace=True)
df.duplicated().sum()

Out[12]:




0

Zero Values
¶

In [ ]:
df.isin([0]).sum()

Out[13]:




id                   0
imdb_id              0
popularity           0
budget            5632
revenue           5947
original_title       0
director             0
runtime             28
genres               0
release_date         0
vote_count           0
vote_average         0
release_year         0
budget_adj        5632
revenue_adj       5947
dtype: int64

looks like we have a lot of zero values specially in budget and revenue and maybe that because a lot of these data are episodes of a TV-sereis, maybe that's why there are no seperate values for budget and revenue for it.

In [ ]:
# Drop zero values then check
df = df[(df != 0).all(1)]

In [ ]:
df.isin([0]).sum()

Out[15]:




id                0
imdb_id           0
popularity        0
budget            0
revenue           0
original_title    0
director          0
runtime           0
genres            0
release_date      0
vote_count        0
vote_average      0
release_year      0
budget_adj        0
revenue_adj       0
dtype: int64

Explore and Visuals
¶

In [ ]:
# check new data
df

Out[16]:
















id


imdb_id


popularity


budget


revenue


original_title


director


runtime


genres


release_date


vote_count


vote_average


release_year


budget_adj


revenue_adj










0


135397


tt0369610


32.985763


150000000


1513528810


Jurassic World


Colin Trevorrow


124


Action|Adventure|Science Fiction|Thriller


2015-06-09


5562


6.5


2015


1.379999e+08


1.392446e+09






1


76341


tt1392190


28.419936


150000000


378436354


Mad Max: Fury Road


George Miller


120


Action|Adventure|Science Fiction|Thriller


2015-05-13


6185


7.1


2015


1.379999e+08


3.481613e+08






2


262500


tt2908446


13.112507


110000000


295238201


Insurgent


Robert Schwentke


119


Adventure|Science Fiction|Thriller


2015-03-18


2480


6.3


2015


1.012000e+08


2.716190e+08






3


140607


tt2488496


11.173104


200000000


2068178225


Star Wars: The Force Awakens


J.J. Abrams


136


Action|Adventure|Science Fiction|Fantasy


20

In [ ]:
df.shape

Out[17]:




(3853, 15)

In [ ]:
df.columns

Out[18]:




Index(['id', 'imdb_id', 'popularity', 'budget', 'revenue', 'original_title',
       'director', 'runtime', 'genres', 'release_date', 'vote_count',
       'vote_average', 'release_year', 'budget_adj', 'revenue_adj'],
      dtype='object')

In [ ]:
df.head(0)

Out[19]:
















id


imdb_id


popularity


budget


revenue


original_title


director


runtime


genres


release_date


vote_count


vote_average


release_year


budget_adj


revenue_adj

In [ ]:
df.describe()

Out[20]:
















id


popularity


budget


revenue


runtime


vote_count


vote_average


release_year


budget_adj


revenue_adj










count


3853.000000


3853.000000


3.853000e+03


3.853000e+03


3853.000000


3853.000000


3853.000000


3853.000000


3.853000e+03


3.853000e+03






mean


39894.523488


1.191825


3.721227e+07


1.077117e+08


109.208928


527.854399


6.168258


2001.259278


4.425038e+07


1.370974e+08






std


67230.100737


1.475258


4.221035e+07


1.765554e+08


19.912913


880.031643


0.795001


11.283517


4.481043e+07


2.161299e+08






min


5.000000


0.001117


1.000000e+00


2.000000e+00


15.000000


10.000000


2.200000


1960.000000


9.693980e-01


2.370705e+00






25%


6073.000000


0.462609


1.000000e+07


1.360940e+07


95.000000


71.000000


5.700000


1995.000000


1.309053e+07


1.840420e+07






50%


11321.000000


0.797723


2.400000e+07


4.480678e+07


106.000000


204.000000


6.200000


2004.000000


3.00

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3853 entries, 0 to 10848
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   id              3853 non-null   int64         
 1   imdb_id         3853 non-null   object        
 2   popularity      3853 non-null   float64       
 3   budget          3853 non-null   int64         
 4   revenue         3853 non-null   int64         
 5   original_title  3853 non-null   object        
 6   director        3853 non-null   object        
 7   runtime         3853 non-null   int64         
 8   genres          3853 non-null   object        
 9   release_date    3853 non-null   datetime64[ns]
 10  vote_count      3853 non-null   int64         
 11  vote_average    3853 non-null   float64       
 12  release_year    3853 non-null   int64         
 13  budget_adj      3853 non-null   float64       
 14  revenue_adj     3853 non-null   float64       
dtypes: 

In [ ]:
# Genral look at the dataframe using hist diagram
df.hist(figsize=(25,25));

Studying Geners
¶

First we are going to seperate Genres into single genre noun
then we will show the schedule and diagram of it

In [ ]:
df.genres

Out[23]:




0        Action|Adventure|Science Fiction|Thriller
1        Action|Adventure|Science Fiction|Thriller
2               Adventure|Science Fiction|Thriller
3         Action|Adventure|Science Fiction|Fantasy
4                            Action|Crime|Thriller
                           ...                    
10822                                        Drama
10828                             Mystery|Thriller
10829                               Action|Western
10835           Action|Adventure|Drama|War|Romance
10848                    Adventure|Science Fiction
Name: genres, Length: 3853, dtype: object

In [ ]:
df['genres'].unique()

Out[24]:




array(['Action|Adventure|Science Fiction|Thriller',
       'Adventure|Science Fiction|Thriller',
       'Action|Adventure|Science Fiction|Fantasy', ...,
       'Action|Drama|Horror|Thriller', 'Adventure|Animation|Drama',
       'Action|Adventure|Drama|War|Romance'], dtype=object)

In [ ]:
df['genres'].nunique()

Out[25]:




1053

looks like Genres are not seperated, so we're going to seperate it first before working on it

Separate Genres to Singles
¶

In [ ]:
# Beginning of seperate genres to study it in better way
# using split function to do that, splitted by '|' 
all_genres = df['genres'].str.split(pat = '|', expand = True)
all_genres.head(1)

Out[26]:
















0


1


2


3


4










0


Action


Adventure


Science Fiction


Thriller


None

shows we have at most 5 Genres in each title got seperated to singles

In [ ]:
# to see the splitted genres in a data frame 
all_genres.head(2)

Out[27]:
















0


1


2


3


4










0


Action


Adventure


Science Fiction


Thriller


None






1


Action


Adventure


Science Fiction


Thriller


None

In [ ]:
# see number of unique genres in each of the 5 columns
all_genres.nunique()

Out[28]:




0    19
1    19
2    19
3    18
4    17
dtype: int64

shows that every column of single genres have at least 17 different genre name

In [ ]:
all_genres.value_counts()

Out[29]:




0          1          2        3                4       
Action     Crime      Drama    Mystery          Thriller    6
           Adventure  Crime    Drama            Thriller    5
Adventure  Animation  Comedy   Family           Fantasy     5
           Action     Comedy   Thriller         Crime       4
Action     Adventure  Comedy   Crime            Thriller    4
                                                           ..
Adventure  Comedy     Fantasy  Science Fiction  Family      1
                      Western  Romance          Thriller    1
           Crime      Drama    Mystery          Western     1
           Drama      Action   Romance          Family      1
Western    Animation  Family   Comedy           Music       1
Length: 209, dtype: int64

In [ ]:
# adding all seperated genres into one single genre to show it properly
g0 = all_genres[0]
g1 = all_genres[1]
g2 = all_genres[2]
g3 = all_genres[3]
g4 = all_genres[4]

# using (append) method to do that
single_genre = g0.append(g1.append(g2.append(g3.append(g4))))

In [ ]:
single_genre

Out[31]:




0           Action
1           Action
2        Adventure
3           Action
4           Action
           ...    
10822         None
10828         None
10829         None
10835      Romance
10848         None
Length: 19265, dtype: object

In [ ]:
#single genre value counts to added it all into seperated genre by value count
single_genre_vc = single_genre.value_counts()

In [ ]:
# the count of single genres
single_genre_vc.count()

Out[33]:




20

looks that we have 20 different Genres

Q- What is the most frequent genre in the date
¶

In [ ]:
#sheduele of single genres frequencies (value counts)
single_genre_vc

Out[34]:




Drama              1755
Comedy             1357
Thriller           1204
Action             1085
Adventure           749
Romance             666
Crime               651
Science Fiction     519
Horror              463
Family              425
Fantasy             396
Mystery             344
Animation           201
Music               136
History             129
War                 119
Western              52
Documentary          35
Foreign              12
TV Movie              1
dtype: int64

In [ ]:
# single genre diagram
single_genre_vc.plot(title= 'Single Genre Count', xlabel= 'Genre', ylabel= 'Count', kind= 'bar', figsize= (8, 8));

A diagram show the counts of titles in each genre

It shows 'Drama' Genre have the most frequency in data

Studying Budget
¶

In [ ]:
df['budget_adj']

Out[36]:




0        1.379999e+08
1        1.379999e+08
2        1.012000e+08
3        1.839999e+08
4        1.747999e+08
             ...     
10822    5.038511e+07
10828    2.015404e+07
10829    3.125892e+07
10835    8.061618e+07
10848    3.436265e+07
Name: budget_adj, Length: 3853, dtype: float64

In [ ]:
#Budget diagram by scatter
scatter_plot_index(df, 'budget_adj', 'budget_adj', 'Genreal look at Budget by Index', 'Budget', 'Index of Budget')

A diagram show the budget by index to get general idea of the budget in the data

In [ ]:
df['budget_adj'].nunique()

Out[38]:




2095

In [ ]:
df['budget_adj'].unique()

Out[39]:




array([1.37999939e+08, 1.01199955e+08, 1.83999919e+08, ...,
       3.12589224e+07, 8.06161763e+07, 3.43626452e+07])

Q- What are the top 10 movies according to Budget
¶

Top 10 Movies by Budget
¶

In [ ]:
df.nlargest(10, 'budget_adj')

Out[40]:
















id


imdb_id


popularity


budget


revenue


original_title


director


runtime


genres


release_date


vote_count


vote_average


release_year


budget_adj


revenue_adj










2244


46528


tt1032751


0.250540


425000000


11087569


The Warrior's Way


Sngmoo Lee


100


Adventure|Fantasy|Action|Western|Thriller


2010-12-02


74


6.4


2010


4.250000e+08


1.108757e+07






3375


1865


tt1298650


4.955130


380000000


1021683000


Pirates of the Caribbean: On Stranger Tides


Rob Marshall


136


Adventure|Action|Fantasy


2011-05-11


3180


6.3


2011


3.683713e+08


9.904175e+08






7387


285


tt0449088


4.965391


300000000


961000000


Pirates of the Caribbean: At World's End


Gore Verbinski


169


Adventure|Fantasy|Action


2007-05-19


2626


6.8


2007


3.155006e+08


1.010654e+09






6570


1452


tt0348150


1.957331


270000000


391081192


Superman Returns


Bryan Singer


154


Adventure|Fantasy|Action|Science Fi

In [ ]:
# Top 10 Movie Titles by Budget
df.nlargest(10, 'budget_adj')[
    ['original_title', 
     'budget_adj']
]

Out[41]:
















original_title


budget_adj










2244


The Warrior's Way


4.250000e+08






3375


Pirates of the Caribbean: On Stranger Tides


3.683713e+08






7387


Pirates of the Caribbean: At World's End


3.155006e+08






6570


Superman Returns


2.920507e+08






5231


Titanic


2.716921e+08






7394


Spider-Man 3


2.713305e+08






1929


Tangled


2.600000e+08






14


Avengers: Age of Ultron


2.575999e+08






1389


Harry Potter and the Half-Blood Prince


2.541001e+08






8089


Waterworld


2.504192e+08

In [ ]:
# Top 10 Budget Movies data diagram
top10_diagram('budget_adj', 'original_title', 'budget_adj', 'Top 10 Movies by Budget', 'Movie Title', 'Budget')

A diagram show the top 10 movies by budget, 'The Warrior's Way' have the highest budget

In [ ]:
# Budget sum of top 10 data by Budget 
df.nlargest(10, 'budget_adj')['budget_adj'].sum()

Out[43]:




2966064259.177309

Q- What is the highest Budget
¶

In [ ]:
df['budget_adj'].max()

Out[44]:




425000000.0

In [ ]:
df['budget_adj'].idxmax()

Out[45]:




2244

In [ ]:
# max movie budget
df[df['budget_adj']==df['budget_adj'].max()]

Out[46]:
















id


imdb_id


popularity


budget


revenue


original_title


director


runtime


genres


release_date


vote_count


vote_average


release_year


budget_adj


revenue_adj










2244


46528


tt1032751


0.25054


425000000


11087569


The Warrior's Way


Sngmoo Lee


100


Adventure|Fantasy|Action|Western|Thriller


2010-12-02


74


6.4


2010


425000000.0


11087569.0

In [ ]:
# max bud movie with title
df.query('index == budget_adj.idxmax()')[['original_title', 'budget_adj']]

Out[47]:
















original_title


budget_adj










2244


The Warrior's Way


425000000.0

Studying Revenue
¶

In [ ]:
df['revenue_adj']

Out[48]:




0        1.392446e+09
1        3.481613e+08
2        2.716190e+08
3        1.902723e+09
4        1.385749e+09
             ...     
10822    2.266436e+08
10828    8.733419e+07
10829    4.030809e+07
10835    1.343603e+08
10848    8.061618e+07
Name: revenue_adj, Length: 3853, dtype: float64

In [ ]:
#revenue diagram by scatter
scatter_plot_index(df, 'revenue_adj', 'revenue_adj', 'Genreal look at Revenue by Index', 'Revenue', 'Index of Revenue')

A diagram shows general idea of the revenue in the data

In [ ]:
df.revenue_adj.nunique()

Out[50]:




3847

In [ ]:
df.revenue_adj.unique()

Out[51]:




array([1.39244589e+09, 3.48161292e+08, 2.71619025e+08, ...,
       4.03080882e+07, 1.34360294e+08, 8.06161763e+07])

Q- What are the top 10 movies according to Revenue
¶

In [ ]:
# Top 10 Movie Titles by Revenue
df.nlargest(10, 'revenue_adj')[
    ['original_title', 
     'revenue_adj']
]

Out[52]:
















original_title


revenue_adj










1386


Avatar


2.827124e+09






1329


Star Wars


2.789712e+09






5231


Titanic


2.506406e+09






10594


The Exorcist


2.167325e+09






9806


Jaws


1.907006e+09






3


Star Wars: The Force Awakens


1.902723e+09






8889


E.T. the Extra-Terrestrial


1.791694e+09






8094


The Net


1.583050e+09






10110


One Hundred and One Dalmatians


1.574815e+09






4361


The Avengers


1.443191e+09

In [ ]:
# Top 10 Revenue Movies data diagram
top10_diagram('revenue_adj', 'original_title', 'revenue_adj', 'Top 10 Movies by Revenue', 'Movie Title', 'Revenue')

A diagram shows the top 10 titles by revenue in the data

In [ ]:
# Revenue sum of top 10 data by revenue 
df.nlargest(10, 'revenue_adj')['revenue_adj'].sum()

Out[54]:




20493045621.538162

Q- What is the highest Revenue
¶

In [ ]:
df['revenue_adj'].max()

Out[55]:




2827123750.41189

In [ ]:
df['revenue_adj'].idxmax()

Out[56]:




1386

In [ ]:
# max movie revenue
df[df['revenue_adj']==df['revenue_adj'].max()]

Out[57]:
















id


imdb_id


popularity


budget


revenue


original_title


director


runtime


genres


release_date


vote_count


vote_average


release_year


budget_adj


revenue_adj










1386


19995


tt0499549


9.432768


237000000


2781505847


Avatar


James Cameron


162


Action|Adventure|Fantasy|Science Fiction


2009-12-10


8458


7.1


2009


2.408869e+08


2.827124e+09

In [ ]:
# max rev movie with title
df.query('index == revenue_adj.idxmax()')[['original_title', 'revenue_adj']]

Out[58]:
















original_title


revenue_adj










1386


Avatar


2.827124e+09

Studying Vote Average
¶

In [ ]:
df['vote_average']

Out[59]:




0        6.5
1        7.1
2        6.3
3        7.5
4        7.3
        ... 
10822    7.5
10828    6.3
10829    6.9
10835    7.0
10848    6.7
Name: vote_average, Length: 3853, dtype: float64

In [ ]:
# Vote_average diagram by scatter
#didn't use plt_scatter_index def because of x and y are opposite to have better view

plt.scatter(y= df['vote_average'], x= df['vote_average'].index)
plt.title('Genreal look at Vote Average by Index')
plt.xlabel('Index of Vote Average')
plt.ylabel('Vote Average')
plt.show()

A diagram shows general idea of the vote average in the data, the highest rate maybe the value of (8)

In [ ]:
df['vote_average'].nunique()

Out[61]:




53

In [ ]:
df['vote_average'].unique()

Out[62]:




array([6.5, 7.1, 6.3, 7.5, 7.3, 7.2, 5.8, 7.6, 8. , 6.2, 5.2, 7.4, 6.1,
       7. , 6.8, 5.3, 7.8, 6.4, 6.6, 7.7, 5.6, 6.9, 5.9, 6.7, 5.5, 5. ,
       4.4, 4.8, 5.1, 5.7, 3.9, 5.4, 4.7, 4.5, 4.9, 4.2, 7.9, 8.2, 6. ,
       3.8, 4.6, 4.1, 4.3, 8.1, 4. , 3.7, 3.3, 8.4, 2.2, 3.4, 3.5, 8.3,
       3.6])

Q- What are top 10 movies according to Rate
¶

Top 10 Movies by Vote Average
¶

In [ ]:
df.nlargest(10, 'vote_average')

Out[63]:
















id


imdb_id


popularity


budget


revenue


original_title


director


runtime


genres


release_date


vote_count


vote_average


release_year


budget_adj


revenue_adj










4178


278


tt0111161


7.192039


25000000


28341469


The Shawshank Redemption


Frank Darabont


142


Drama|Crime


1994-09-10


5754


8.4


1994


3.677779e+07


4.169346e+07






7948


24128


tt0088178


0.283191


1200000


4978922


Stop Making Sense


Jonathan Demme


88


Documentary|Music


1984-11-16


23


8.4


1984


2.518850e+06


1.045097e+07






7269


238


tt0068646


5.738034


6000000


245066411


The Godfather


Francis Ford Coppola


175


Drama|Crime


1972-03-15


3970


8.3


1972


3.128737e+07


1.277914e+09






650


244786


tt2582802


4.780419


3300000


13993093


Whiplash


Damien Chazelle


105


Drama|Music


2014-10-10


2372


8.2


2014


3.039600e+06


1.288891e+07






2409


550


tt0137523


8.947905


63000000


10085375

In [ ]:
# Top 10 Movie Titles by Rate
df.nlargest(10, 'vote_average')[
    ['original_title', 
     'vote_average']
]

Out[64]:
















original_title


vote_average










4178


The Shawshank Redemption


8.4






7948


Stop Making Sense


8.4






7269


The Godfather


8.3






650


Whiplash


8.2






2409


Fight Club


8.1






2875


The Dark Knight


8.1






4177


Pulp Fiction


8.1






4179


Forrest Gump


8.1






9758


The Godfather: Part II


8.1






10222


Schindler's List


8.1

In [ ]:
# Top 10 Vote Average Movies data diagram
top10_diagram('vote_average', 'original_title', 'vote_average', 'Top 10 Movies by Vote Average', 'Movie Title', 'Vote Average')

A diagram shows the top 10 titles by vote average in the data

In [ ]:
# Average vote of top 10 movies by vote average
df.nlargest(10, 'vote_average')['vote_average'].mean()

Out[66]:




8.19

Q- What is the highest title rate
¶

In [ ]:
df['vote_average'].max()

Out[67]:




8.4

In [ ]:
df['vote_average'].idxmax()

Out[68]:




4178

In [ ]:
# max title Rate
df[df['vote_average']==df['vote_average'].max()]

Out[69]:
















id


imdb_id


popularity


budget


revenue


original_title


director


runtime


genres


release_date


vote_count


vote_average


release_year


budget_adj


revenue_adj










4178


278


tt0111161


7.192039


25000000


28341469


The Shawshank Redemption


Frank Darabont


142


Drama|Crime


1994-09-10


5754


8.4


1994


3.677779e+07


4.169346e+07






7948


24128


tt0088178


0.283191


1200000


4978922


Stop Making Sense


Jonathan Demme


88


Documentary|Music


1984-11-16


23


8.4


1984


2.518850e+06


1.045097e+07

In [ ]:
# max rate movie with title
df.query('index == vote_average.idxmax()')[['original_title', 'vote_average']]

Out[70]:
















original_title


vote_average










4178


The Shawshank Redemption


8.4

It shows that "The Shawshank Redemption" with (8.4) rate is the highest

vote average category
¶

Seperate vote average data into four categories using describe function to know the limits

In [ ]:
df['vote_average'].describe()

Out[71]:




count    3853.000000
mean        6.168258
std         0.795001
min         2.200000
25%         5.700000
50%         6.200000
75%         6.700000
max         8.400000
Name: vote_average, dtype: float64

In [ ]:
# vote average bigger or equal to 6.7 
vah = df.query('vote_average >= 6.7')
vah_un = vah['vote_average'].unique()
vah_nu = vah['vote_average'].nunique() 
vah_c = vah['vote_average'].count() 

# bigger or equal to 6.2 
vam = df.query('vote_average >= 6.2')
vam_un = vam['vote_average'].unique()
vam_nu = vam['vote_average'].nunique() 
vam_c = vam['vote_average'].count() 

# bigger or equal to 5.7
valw = df.query('vote_average >= 5.7')
valw_un = valw['vote_average'].unique()
valw_nu = valw['vote_average'].nunique() 
valw_c = valw['vote_average'].count() 

# less than 5.7
vavl = df.query('vote_average < 5.7')
vavl_un = vavl['vote_average'].unique()
vavl_nu = vavl['vote_average'].nunique() 
vavl_c = vavl['vote_average'].count()

In [ ]:
# vote average unique
vaun = vah_un, vam_un, valw_un, vavl_un
# vote average category count
vacc = vah_c, vam_c, valw_c, vavl_c

In [ ]:
vaun

Out[74]:




(array([7.1, 7.5, 7.3, 7.2, 7.6, 8. , 7.4, 7. , 6.8, 7.8, 7.7, 6.9, 6.7,
        7.9, 8.2, 8.1, 8.4, 8.3]),
 array([6.5, 7.1, 6.3, 7.5, 7.3, 7.2, 7.6, 8. , 6.2, 7.4, 7. , 6.8, 7.8,
        6.4, 6.6, 7.7, 6.9, 6.7, 7.9, 8.2, 8.1, 8.4, 8.3]),
 array([6.5, 7.1, 6.3, 7.5, 7.3, 7.2, 5.8, 7.6, 8. , 6.2, 7.4, 6.1, 7. ,
        6.8, 7.8, 6.4, 6.6, 7.7, 6.9, 5.9, 6.7, 5.7, 7.9, 8.2, 6. , 8.1,
        8.4, 8.3]),
 array([5.2, 5.3, 5.6, 5.5, 5. , 4.4, 4.8, 5.1, 3.9, 5.4, 4.7, 4.5, 4.9,
        4.2, 3.8, 4.6, 4.1, 4.3, 4. , 3.7, 3.3, 2.2, 3.4, 3.5, 3.6]))

In [ ]:
# vote average category names
#vacnm = "High", "Medium", "Low", "Very Low"

In [ ]:
# vote average category names
vacnm = "More than (6.7)", "More than (6.2)", "More than (5.7)", "Less than (5.7)"

In [ ]:
vacnm

Out[77]:




('More than (6.7)', 'More than (6.2)', 'More than (5.7)', 'Less than (5.7)')

In [ ]:
#vote average category counts for each category
vacc

Out[78]:




(1062, 2012, 2895, 958)

In [ ]:
#the scheduele
#making a dataframe

c1 = vacnm
c2 = vacc
s = c1, c2

vac_sch = pd.DataFrame(list(s), index = ['category', 'count'])
vac_sch = vac_sch.T

Rate category count scheduele
¶

In [ ]:
vac_sch

Out[80]:
















category


count










0


More than (6.7)


1062






1


More than (6.2)


2012






2


More than (5.7)


2895






3


Less than (5.7)


958

Rate category diagram
¶

In [ ]:
x = vacnm
y = vacc

plt.title("Category of Vote Average")
plt.ylabel("Rate")
plt.xlabel("Count")

plt.barh(x, y)

plt.show()

A diagram shows the category of vote average seperated into 4 categories, the highest is 'More than (6.7)', the lowest is 'Less than (5.7)' in the data

Q- Who is the highest rated director
¶

In [ ]:
df['vote_average'].idxmax()

Out[82]:




4178

In [ ]:
df.groupby(by = 'director').mean()['vote_average'].idxmax()

Out[83]:




'Damien Chazelle'

In [ ]:
df.groupby(by = 'director').mean().query('director == "Damien Chazelle"')['vote_average']

Out[84]:




director
Damien Chazelle    8.2
Name: vote_average, dtype: float64

it shows that Damien Chazelle is best director according to vote average

Relations
¶

Q- How Budget affect Rate
¶

In [ ]:
scatter_plot(df, 'budget_adj', 'vote_average', 'Budget by Vote Average', 'Budget', 'Vote Average', '#684579')

normal correlation between Budget and rate not so strongly positive

Q- How Revenue affect Rate
¶

In [ ]:
scatter_plot(df, 'revenue_adj', 'vote_average', 'Revenue by Vote Average', 'Revenue', 'Vote Average', '#FC600A')

slight positive correlation between revenue and rate not so strong

Conclusion
¶

Limitations
¶


- The dropped rows like null or zero values may have some valuable data that may make the analysis better and strong             results. 

- There’re no data to separate (TV Shows) or (Movies) or other shows, if there’s a column that have this data, will be much helpful to get better and clear conclusions. 

- Missing (awards) data, adding that would be helpful to get a better idea to analyze the data in subjects like the show, directors or cast. 

- Missing the currency in numerus data like budget and revenue that may give a lot of misunderstanding to the data, especially to get benefit from this project in business fields. 

- The missing of the category of the show (movie), (series), etc. gave a lack of information like the episode numbers, may affect the conclusions of the project.

Conclusions
¶


- Drama is the most frequent genre followed by Comedy then Thriller.

- “The Warrior's Way” is the Highest budget movie with 4.25e+08. 
- Total budget of top 10 movies equals to 2.96e+09.

- “Avatar” is the Highest revenue with 2.83e+09.
- Total revenue of top 10 movies equals to 20.5e+09.

- “The Shawshank Redemption” have the highest rate equals to 8.4
- Average vote of top 10 movies by vote average is 8.19.

- The correlation between rate and budget is normal and increasing positively a little. 
- The correlation between rate and revenue is slightly positive but not so strong.